src/GraphQl/Type/SchemaBuilder.php

```diff
@@ -155,20 +173,30 @@ private function getNodeQueryField(): array
 
     /**
      * Gets the query fields of the schema.
+     *
+     * @param array|bool $itemConfiguration false if not configured
+     * @param array|bool $collectionConfiguration false if not configured
      */
-    private function getQueryFields(string $resourceClass, ResourceMetadata $resourceMetadata): array
+    private function getQueryFields(string $resourceClass, ResourceMetadata $resourceMetadata, string $operationName = 'query', $itemConfiguration, $collectionConfiguration): array
     {
         $queryFields = [];
         $shortName = $resourceMetadata->getShortName();
-        $deprecationReason = $resourceMetadata->getGraphqlAttribute('query', 'deprecation_reason', '', true);
+        $fieldName = lcfirst('query' === $operationName ? $shortName : $operationName.$shortName);
 
-        if ($fieldConfiguration = $this->getResourceFieldConfiguration($resourceClass, $resourceMetadata, null, $deprecationReason, new Type(Type::BUILTIN_TYPE_OBJECT, true, $resourceClass), $resourceClass)) {
-            $fieldConfiguration['args'] += ['id' => ['type' => GraphQLType::id()]];
-            $queryFields[lcfirst($shortName)] = $fieldConfiguration;
+        $deprecationReason = $resourceMetadata->getGraphqlAttribute($operationName, 'deprecation_reason', '', true);
+
+        if ($itemConfiguration && $fieldConfiguration = $this->getResourceFieldConfiguration($resourceClass, $resourceMetadata, null, $deprecationReason, new Type(Type::BUILTIN_TYPE_OBJECT, true, $resourceClass), $resourceClass)) {
+            if (\is_array($itemConfiguration)) {
+                $fieldConfiguration = array_merge($fieldConfiguration, $itemConfiguration);
```

Comment:
```suggestion
                $fieldConfiguration += $itemConfiguration;
```

Reply:
This does not result in the same behaviour, in case of keys existing in both arrays it would take those of the `$fieldConfiguration` (the "generated" ones). I think it should be able to override them using `$itemConfiguration`, which is why I chose `array_merge`.